connectors/spark/src/main/scala/io/unitycatalog/spark/UCSingleCatalog.scala

```diff
@@ -155,6 +161,66 @@ class UCSingleCatalog extends TableCatalog with SupportsNamespaces with Logging
   override def dropNamespace(namespace: Array[String], cascade: Boolean): Boolean = {
     delegate.asInstanceOf[DelegatingCatalogExtension].dropNamespace(namespace, cascade)
   }
+
+  override def stageCreate(ident: Identifier, columns: Array[Column], partitions: Array[Transform], properties: util.Map[String, String]): StagedTable = {
+    val oldProperties = loadTableProperties(ident, properties)
+    val newTable = super.createTable(ident, columns, partitions, oldProperties ++ properties)
+    BestEffortStagedTable(ident, Option(newTable).getOrElse(loadTable(ident)), this)
+  }
+
+  override def stageReplace(ident: Identifier, columns: Array[Column], partitions: Array[Transform], properties: util.Map[String, String]): StagedTable = {
+    val oldProperties = loadTableProperties(ident, properties)
+    super.dropTable(ident)
+    val newTable = super.createTable(ident, columns, partitions, oldProperties ++ properties)
+    BestEffortStagedTable(ident, Option(newTable).getOrElse(loadTable(ident)), this)
+  }
+
+  override def stageCreateOrReplace(ident: Identifier, columns: Array[Column], partitions: Array[Transform], properties: util.Map[String, String]): StagedTable = {
+    val oldProperties = loadTableProperties(ident, properties)
+    try super.dropTable(ident)
+    catch {
+      case _: NoSuchTableException => // this is fine
+    }
+    val newTable = super.createTable(ident, columns, partitions, oldProperties ++ properties)
+    BestEffortStagedTable(ident, Option(newTable).getOrElse(loadTable(ident)), this)
+  }
+
+  def loadTableProperties(ident: Identifier, properties: util.Map[String, String]): util.Map[String, String] = {
+    if (ucProxy != null && ucProxy == this.delegate &&
```

Comment:
We don't need this new `ucProxy` variable, we already have `UCSingleCatalog.DELTA_CATALOG_LOADED`

Reply:
done